eng/common/TestResources/New-TestResources.ps1

```
@@ -123,6 +122,23 @@ if (!$templateFiles) {
     exit
 }
 
+# If no location is specified use safe default locations for the given
+# environment. If no matching environment is found $Location remains an empty
+# string.
+if (!$Location) {
+    $defaultLocations = @{
+        'AzureCloud' = 'westus2';
+        'AzureUSGovernment' = 'usgovvirginia';
+        'AzureChinaCloud' = 'chinaeast2';
+    }
+
+    if ($defaultLocations.ContainsKey($Environment)) {
+        $Location = $defaultLocations[$Environment]
+    }
+
+    Write-Verbose "Location was not set. Using default location for environment: '$Location'"
```

You should throw an error if the location is still empty.